scripts/sanitise-db.sql

```diff
@@ -260,30 +268,44 @@ CREATE OR REPLACE FUNCTION anonymize_risk_register_table() RETURNS VOID AS $$
 DECLARE
     row_count INTEGER;
     pre_update_count INTEGER;
-    lorem_text TEXT;
+    lorem_full_desc TEXT;
+    lorem_risk_name TEXT;
+    lorem_short_desc TEXT;
+    lorem_mitigations TEXT;
+    lorem_consequences TEXT;
+    lorem_risk_owner_role TEXT;
 BEGIN
     RAISE NOTICE 'Anonymising multiple DATA_BLOB keys in RISK_REGISTER table';
 
     SELECT COUNT(*) INTO pre_update_count FROM risk_register WHERE data_blob IS NOT NULL;
     RAISE NOTICE 'Number of rows to be updated: %', pre_update_count;
 
-    lorem_text := anon.lorem_ipsum(words := 30);
+    lorem_full_desc := anon.lorem_ipsum(words := 30);
+    lorem_risk_name := anon.lorem_ipsum(words := 10);
+    lorem_short_desc := anon.lorem_ipsum(words := 5);
+    lorem_mitigations := anon.lorem_ipsum(words := 8);
+    lorem_consequences := anon.lorem_ipsum(words := 6);
+    lorem_risk_owner_role := anon.lorem_ipsum(words := 3);
 
-    BEGIN
-        UPDATE risk_register
-        SET data_blob = jsonb_set(
-            data_blob,
-            '{}',
-            jsonb_build_object(
-                'full_desc', substring(lorem_text from 1 for 20),
-                'risk_name', substring(lorem_text from 20 for 22),
-                'short_desc', substring(lorem_text from 10 for 16),
-                'mitigations', substring(lorem_text from 1 for 10),
-                'consequences', substring(lorem_text from 1 for 5),
-                'risk_owner_role', substring(lorem_text from 1 for 3)
-            )
-        )
-        WHERE data_blob IS NOT NULL;
+BEGIN
+UPDATE risk_register
+SET data_blob = jsonb_set(
```

Comment:
So the idea here is to only update specific fields, leaving unnamed fields unsanitised? Defaulting to unsanitised might not be the best approach in terms of future-proofing the effectiveness of this sanitisation process, as any new fields with PII that are added to the database but are not added here will be visible as-is in the supposedly sanitised output.

If we're okay with this, then a more concise approach could involve using [JSON object concatenation](https://stackoverflow.com/questions/30101603/merging-concatenating-jsonb-columns-in-query) and in-lining the calls to `anon.lorem_ipsum` when building the new JSON object to concatenate the existing one with, e.g.,:

```
new_values := jsonb_build_object(
   'full_desc', substring(anon.lorem_ipsum(words := 50) from 1 for 200),
   'risk_name', substring(anon.lorem_ipsum(words := 10) from 1 for 50),
   'short_desc', substring(anon.lorem_ipsum(words := 20) from 1 for 100),
   'mitigations', substring(anon.lorem_ipsum(words := 30) from 1 for 150),
   'consequences', substring(anon.lorem_ipsum(words := 30) from 1 for 150),
   'risk_owner_role', substring(anon.lorem_ipsum(words := 5) from 1 for 50)
);
```

And then:
```
UPDATE risk_register
SET data_blob = data_blob || new_values
WHERE data_blob IS NOT NULL;
```

Reply:
Yeah that's the pain point and we might think how we will manage the newly coming PII but for now will need to add here if something new comes with the PII. 

In the previous we are using `jsonb_build_object` and it was not sanitising working but yes this looks fine. Thank You.